samples/nrf9160/nrf_cloud_rest_agps/src/main.c

```diff
@@ -0,0 +1,735 @@
+			(void)k_sem_reset(&lte_connected);
+			disconnected = true;
+			(void)set_led(0);
+			break;
+		}
+		disconnected = false;
+
+		LOG_INF("Network registration status: %s",
+			evt->nw_reg_status == LTE_LC_NW_REG_REGISTERED_HOME ?
+			"Connected - home network" : "Connected - roaming");
+		k_sem_give(&lte_connected);
+		break;
+	case LTE_LC_EVT_CELL_UPDATE:
+		LOG_DBG("LTE cell changed: Cell ID: %d, Tracking area: %d",
+			evt->cell.id, evt->cell.tac);
+		current_cell.id = evt->cell.id;
+		current_cell.tac = evt->cell.tac;
+		break;
+	case LTE_LC_EVT_LTE_MODE_UPDATE:
+	case LTE_LC_EVT_PSM_UPDATE:
+	case LTE_LC_EVT_EDRX_UPDATE:
+	case LTE_LC_EVT_RRC_UPDATE:
+	default:
+		break;
+	}
+}
+
+static int connect_to_network(void)
+{
+	int err;
+
+	LOG_INF("Waiting for network..");
+
+	(void)k_sem_reset(&lte_connected);
+
+#if defined(CONFIG_LTE_POWER_SAVING_MODE)
+	err = lte_lc_psm_req(true);
+	if (err) {
+		LOG_ERR("PSM request failed, error: %d", err);
+		return err;
+	}
+	LOG_INF("PSM mode requested");
+#endif
+
+	err = lte_lc_init_and_connect_async(lte_handler);
+	if (err) {
+		LOG_ERR("Failed to init modem, err %d", err);
+	} else {
+		err = wait_connected();
+	}
+
+	return err;
+}
+
+static int wait_connected(void)
+{
+	int err;
+
+	do {
+		err = k_sem_take(&lte_connected, K_SECONDS(10));
+		if (err == -EAGAIN) {
+			LOG_WRN("Timeout waiting for LTE: asking to connect...");
+			err = lte_lc_connect();
+			if (err) {
+				LOG_ERR("Error on lte_lc_connect():%d", err);
+				failure_count++;
+			}
+		}
+	} while (err);
+
+	LOG_INF("Connected");
+	(void)set_led(1);
+
+	return err;
+}
+
+void main(void)
+{
+	int err;
+	uint64_t fix_timestamp = 0;
+
+	cls_home();
+	LOG_INF("Starting nRF Cloud REST A-GPS API Sample");
+
+	err = init();
+	if (err) {
+		LOG_ERR("Initialization failed");
+		failure_count = FAILURE_LIMIT;
+		goto error;
+	}
+
+	err = connect_to_network();
+	if (err) {
+		failure_count++;
+	}
+
+	err = init_gnss();
+	if (err) {
+		failure_count = FAILURE_LIMIT;
+		goto error;
+	}
+	LOG_INF("GPS started");
+
+	agps_print_enable(false);
+
+	(void)do_jitp();
+
+	for (; failure_count < FAILURE_LIMIT;) {
+		(void)k_poll(events, NUM_EVENTS, K_FOREVER);
+
+		if (events[0].state == K_POLL_STATE_SEM_AVAILABLE &&
+		    k_sem_take(events[0].sem, K_NO_WAIT) == 0) {
+			/* New PVT data available */
+
+			events[0].state = K_POLL_STATE_NOT_READY;
+			if (!IS_ENABLED(CONFIG_GPS_SAMPLE_NMEA_ONLY) &&
+			    !agps_data_download_ongoing()) {
+				print_satellite_stats(&last_pvt);
+
+				if (gnss_blocked) {
+					LOG_INF("GNSS operation blocked by LTE");
+				}
+				LOG_INF("---------------------------------\n");
+
+				if (last_pvt.flags & NRF_MODEM_GNSS_PVT_FLAG_FIX_VALID) {
+					fix_timestamp = k_uptime_get();
+					print_pvt_data(&last_pvt);
+				} else {
+					LOG_INF("Seconds since last fix: %lld",
+							(k_uptime_get() - fix_timestamp) / 1000);
+				}
+#if defined(CONFIG_REST_NMEA)
+				LOG_INF("NMEA strings:");
```

Comment:
should this be moved into the NMEA processing block below where the string is actually printed?

Reply:
This is copied nearly verbatim from the GPS sample.  It appears to be intentional -- for every PVT message, there can be multiple NMEA strings immediately thereafter, so printing it this way ensures they are printed together right after a single copy of "NMEA strings:" is printed.